backend/src/govsg/middlewares/govsg-template.middleware.ts

```diff
@@ -21,19 +23,47 @@ import { NextFunction, Request, Response } from 'express'
 const logger = loggerWithLabel(module)
 
 export const getAvailableTemplates = async (
-  _req: Request,
+  req: Request,
   res: Response
 ): Promise<Response> => {
-  const templates = await GovsgTemplate.findAll()
+  if (!req.session?.user?.id) {
+    throw new ApiAuthenticationError('Request not authenticated')
+  }
+  const userId = req.session.user.id
+  const allowedTemplateIds = await GovsgTemplatesAccess.findAll({
+    where: { userId },
+  }).then((templates) => templates.map((template) => template.templateId))
+  const templates = await GovsgTemplate.findAll({
+    where: { id: allowedTemplateIds },
+  })
   return res.status(200).json({ data: templates })
 }
 
+async function checkUserTemplateAccess(
+  userId: number,
+  templateId: number
+): Promise<void> {
+  const allowed = await GovsgTemplatesAccess.findOne({
+    where: { userId, templateId },
+  })
+  if (!allowed) {
+    throw new ApiInvalidTemplateError(
+      `User ${userId} does not have access to template ${templateId}`
+    )
+  }
+}
+
 export async function pickTemplateForCampaign(
   req: Request,
   res: Response
 ): Promise<Response> {
+  if (!req.session?.user?.id) {
```

Comment:
no need to add this, we already have an authentication middle ware installed in the route iinm

Reply:
yeah it's not necessary but I was just following [the code here hmm](https://github.com/opengovsg/postmangovsg/blob/dc4785a7c1f53972ad181537a57c580df2d603d8/backend/src/core/middlewares/experiment.middleware.ts)

Reply:
oh lmao yea let's remove both

Reply:
OK done